packages/text-annotator-react/src/TextAnnotator.tsx

```diff
@@ -1,13 +1,16 @@
 import { ReactNode, useContext, useEffect, useRef } from 'react';
 import { AnnotoriousContext, DrawingStyle, Filter } from '@annotorious/react';
+import type { FormatAdapter } from '@annotorious/core';
 import type { TextAnnotation, TextAnnotatorOptions } from '@recogito/text-annotator';
 import { createTextAnnotator } from '@recogito/text-annotator';
 
 import '@recogito/text-annotator/dist/text-annotator.css';
 
-export type TextAnnotatorProps<E extends unknown> = TextAnnotatorOptions<E> & {
+export interface TextAnnotatorProps<E extends unknown> extends Omit<TextAnnotatorOptions<E>, 'adapter'>  {
 
-  children?: ReactNode | JSX.Element; 
+  children?: ReactNode | JSX.Element;
+
+  adapter?: FormatAdapter<TextAnnotation, E> | ((container: HTMLElement) => FormatAdapter<TextAnnotation, E>) | null;
```

Comment:
1. W/o using the `interfaced`  Typescript still points to the `adapter` prop from the `TextAnnotatorOptions`, not the overridden one
2. `adapter` is overridden only within the `TextAnnotator.tsx`, because only the React wrapper automatically creates a container element:
https://github.com/recogito/text-annotator-js/blob/9dbac6590ed5c2936c7bb28a5739b8f06209d3f3/packages/text-annotator-react/src/TextAnnotator.tsx#L48-L52